src/main/java/org/jenkinsci/plugins/fodupload/polling/ScanStatusPoller.java

```diff
@@ -201,3 +169,97 @@ private void printPauseMessages(ScanSummaryDTO scanSummary) {
         }
     }
 }
+
+class StatusPollerThread extends Thread {
+    public Boolean fail = false;
+    public Boolean finished = false;
+    public Boolean cancelled = false;
+    public Boolean waiting = false;
+    public String statusString;
+    public PollReleaseStatusResult result = new PollReleaseStatusResult();
+    public ScanSummaryDTO scanSummaryDTO = null;
+    public ReleaseDTO releaseDTO = null;
+    public PrintStream logger;
+    private int releaseId;
+    private int pollingInterval;
+    private ReleaseController releaseController = null;
+    private StaticScanSummaryController scanSummaryController;
+    private List<LookupItemsModel> analysisStatusTypes = null;
+    private List<String> completeStatusList = null;
+
+
+    StatusPollerThread(String name, final int releaseId, List<LookupItemsModel> analysisStatusTypes,
+                       FodApiConnection apiConnection, List<String> completeStatusList, PrintStream logger, int pollingInterval) {
+        super(name);
+        this.releaseId = releaseId;
+        this.analysisStatusTypes = analysisStatusTypes;
+        this.logger = logger;
+        this.releaseController = new ReleaseController(apiConnection);
+        this.scanSummaryController = new StaticScanSummaryController(apiConnection, logger);
+        this.completeStatusList = completeStatusList;
+        this.pollingInterval = pollingInterval;
+    }
+
+    public void run() {
+        try {
+            Thread.sleep(1000 * 60 * this.pollingInterval);
```

Comment:
We probably need to handle the `InterruptedException` here.  I was curious how a long running process on a managed thread could be interrupted by jdk.. found some reasoning about that here: https://www.yegor256.com/2015/10/20/interrupted-exception.html

Reply:
I've been looking through the link and considering design changes. The thread has a finite set of tasks organized in a serial way with a clear exit condition. I'm not sure if this will actually impact here.

Reply:
The main thread that has that while loop could benefit from this OR if I abstracted that main thread out as well...

Reply:
Wrapped the main thread in a try catch for Interrupted exceptions and changed the exception on the thread's try catch.